CondCore/Utilities/python/tier0.py

```diff
@@ -90,43 +90,60 @@ def unsetDebug( self ):
     def setProxy( self, proxy ):
         self._proxy = proxy
 
-    def _queryTier0DataSvc( self, url ):
-        """
-        Queries Tier0DataSvc.
-        url: Tier0DataSvc URL.
-        @returns: dictionary, from whence the required information must be retrieved according to the API call.
-        Raises if connection error, bad response, or timeout after retries occur.
-        """
+    def _getCerts( self ) -> str:
+        cert_path = os.getenv('X509_USER_CERT', '')
+        key_path = os.getenv('X509_USER_KEY', '')
         
-        userAgent = "User-Agent: ConditionWebServices/1.0 python/%d.%d.%d PycURL/%s" % ( sys.version_info[ :3 ] + ( pycurl.version_info()[ 1 ], ) )
+        certs = ""
+        if cert_path:
+            certs += f' --cert {cert_path}'
+        else:
+            logging.warning("No certificate, nor proxy provided for Tier0 access")
+        if key_path:
+            certs += f' --key {key_path}'
+        return certs
+
+    def _curlQueryTier0( self, url:str, force_debug:bool = False):
+        userAgent = "User-Agent: ConditionWebServices/1.0 python/%d.%d.%d PycURL/%s" \
+            % ( sys.version_info[ :3 ] + ( pycurl.version_info()[ 1 ], ) )
+        debug = "-v" if self._debug or force_debug else "-s -S"
 
         proxy = ""
-        if self._proxy: proxy = ' --proxy=%s ' % self._proxy
-        
-        debug = " -s -S "
-        if self._debug: debug = " -v "
-        
-        cmd = '/usr/bin/curl -k -L --user-agent "%s" %s --connect-timeout %i --retry %i %s %s ' % (userAgent, proxy, self._timeOut, self._retries, debug, url)
+        certs = ""
+        if self._proxy:
+            proxy = f"--proxy {self._proxy}"
+        else:
+            certs = self._getCerts()
```

Comment:
I would like to spark a discussion if this is the correct way to do it. We discussed it with @vlimant that this seems to be the safest option for an immediate fix. The idea is to provide the cert and key only when the proxy is not set so we don't disrupt already working uses of this API that are authorized through a proxy. 
But this is based on a few assumptions that I'm not sure of:
- this proxy is used for authorization
- there are no use cases needing both proxy and certs (seems untrue or like it can become untrue and something will break
- providing both proxy and certs might result in an error or something unexpected

Maybe when a proxy is provided we should first try without the certs and if it fails we should provide the certs?

Reply:
There's already a mechanism of retrying a failed curl request here so I've just made it provide both cert and proxy on the second try (if a proxy is used and the request with the proxy without cert failed). I think it's the most foolproof solution